sx126x/sx126x.go

```diff
@@ -0,0 +1,714 @@
+		d.SetSyncWord(SX126X_LORA_MAC_PRIVATE_SYNCWORD)
+	}
+}
+
+// SetPacketParam sets various packet-related params
+func (d *Device) SetPacketParam(preambleLength uint16, headerType, crcType, payloadLength, invertIQ uint8) {
+	var p [6]uint8
+	p[0] = uint8((preambleLength >> 8) & 0xFF)
+	p[1] = uint8(preambleLength & 0xFF)
+	p[2] = headerType
+	p[3] = payloadLength
+	p[4] = crcType
+	p[5] = invertIQ
+	d.ExecSetCommand(SX126X_CMD_SET_PACKET_PARAMS, p[:])
+}
+
+// SetBufferBaseAddress sets base address for buffer
+func (d *Device) SetBufferBaseAddress(txBaseAddress, rxBaseAddress uint8) {
+	var p [2]uint8
+	p[0] = txBaseAddress
+	p[1] = rxBaseAddress
+	d.ExecSetCommand(SX126X_CMD_SET_BUFFER_BASE_ADDRESS, p[:])
+}
+
+// SetRfFrequency sets the radio frequency
+func (d *Device) SetRfFrequency(frequency uint32) {
+	var p [4]uint8
+	freq := uint32((uint64(frequency) << 25) / 32000000)
+	p[0] = uint8((freq >> 24) & 0xFF)
+	p[1] = uint8((freq >> 16) & 0xFF)
+	p[2] = uint8((freq >> 8) & 0xFF)
+	p[3] = uint8((freq >> 0) & 0xFF)
+	d.ExecSetCommand(SX126X_CMD_SET_RF_FREQUENCY, p[:])
+}
+
+// SetCurrentLimit sets max current in the module
+func (d *Device) SetCurrentLimit(limit uint8) {
+	if limit > 140 {
+		limit = 140
+	}
+	rawLimit := uint8(float32(limit) / 2.5)
+	p := []uint8{rawLimit}
+	d.WriteRegister(SX126X_REG_OCP_CONFIGURATION, p[:])
+}
+
+// SetTxConfig sets power and rampup time
+func (d *Device) SetTxParams(power int8, rampTime uint8) {
+	var p [2]uint8
+
+	if d.deviceType == DEVICE_TYPE_SX1261 {
+		if power == 15 {
+			d.SetPaConfig(0x06, 0x00, 0x01, 0x01)
+		} else {
+			d.SetPaConfig(0x04, 0x00, 0x01, 0x01)
+		}
+		if power > 14 {
+			power = 14
+		} else if power < -3 {
+			power = -3
+		}
+		d.SetCurrentLimit(80) // Set max current limit to 80mA
+	} else { // sx1262 and sx1268
+		d.SetPaConfig(0x04, 0x07, 0x00, 0x01)
+		if power > 22 {
+			power = 22
+		} else if power < -3 {
+			power = -3
+		}
+		d.SetCurrentLimit(140) // Set max current limit to 140 mA
+	}
+
+	p[0] = uint8(power)
+	p[1] = rampTime
+	d.ExecSetCommand(SX126X_CMD_SET_TX_PARAMS, p[:])
+}
+
+// SetModulationParams sets the Lora modulation frequency
+func (d *Device) SetModulationParams(spreadingFactor, bandwidth, codingRate, lowDataRateOptimize uint8) {
+	var p [4]uint8
+	p[0] = spreadingFactor
+	p[1] = bandwidth
+	p[2] = codingRate
+	p[3] = lowDataRateOptimize
+	d.ExecSetCommand(SX126X_CMD_SET_MODULATION_PARAMS, p[:])
+}
+
+// CheckDeviceReady sleep until all busy flags clears
+func (d *Device) CheckDeviceReady() error {
+	if d.deepSleep == true {
+		d.SpiSetNss(false)
+		time.Sleep(time.Millisecond)
+		d.SpiSetNss(true)
+		d.deepSleep = false
+	}
+	return d.WaitBusy()
+}
+
+// ExecSetCommand send a command to configure the peripheral
+func (d *Device) ExecSetCommand(cmd uint8, buf []uint8) {
+	d.CheckDeviceReady()
+	if cmd == SX126X_CMD_SET_SLEEP {
+		d.deepSleep = true
+	} else {
+		d.deepSleep = false
+	}
+	d.SpiSetNss(false)
+	// Send command and params
+	d.SpiTx(cmd)
```

Comment:
Why not use this

```go
d.spi.Tx(append(cmd, buf), nil)
```

instead of this block of lines?

Reply:
Ok, 
I could removed the SpiTx 'wrapper' I was using. 
I could then use  s.spi.Tx and remove the loops.

Reply:
I think this change looks a lot cleaner, thank you for making it!